front/app/containers/Admin/projects/project/nativeSurvey/FormResults/FormResultsQuestion/TextQuestion/Analysis/index.tsx

```diff
@@ -1,211 +1,55 @@
-
-import { updateSearchParams } from 'utils/cl-router/updateSearchParams';
-import ConsentModal from './ConsentModal';
-import CreateAnalysisModal from './CreateAnalysisModal';
 import useAnalyses from 'api/analyses/useAnalyses';
-import useDeleteAnalysis from 'api/analyses/useDeleteAnalysis';
 import { useParams } from 'react-router-dom';
 
-import Button from 'components/UI/Button';
-import useFormCustomFields from 'api/custom_fields/useCustomFields';
-import { isNilOrError } from 'utils/helperUtils';
-import tracks from 'containers/Admin/projects/project/analysis/tracks';
-import { trackEventByName } from 'utils/analytics';
-import clHistory from 'utils/cl-router/history';
-import Divider from 'components/admin/Divider';
 import AnalysisInsights from './AnalysisInsights';
+import useAddAnalysis from 'api/analyses/useAddAnalysis';
 
 const Analysis = ({ customFieldId }: { customFieldId: string }) => {
-  const [dropdownIsOpened, setDropdownIsOpened] = useState(false);
-  const { formatMessage } = useIntl();
+  const { mutate: addAnalysis } = useAddAnalysis();
+
   const { projectId, phaseId } = useParams() as {
     projectId: string;
     phaseId: string;
   };
 
-  const { mutate: deleteAnalysis } = useDeleteAnalysis();
   const { data: analyses } = useAnalyses({
     projectId: phaseId ? undefined : projectId,
     phaseId,
   });
 
-  const { data: formCustomFields } = useFormCustomFields({
-    projectId,
-    phaseId,
-  });
-
-  const handleDeleteAnalysis = (analysisId: string) => {
-    if (window.confirm(formatMessage(messages.deleteAnalysisConfirmation))) {
-      deleteAnalysis(analysisId, {
-        onSuccess: () => {
-          trackEventByName(tracks.analysisForSurveyDeleted.name, {
-            extra: { projectId },
-          });
-        },
-      });
-    }
-  };
-  const [consentModalIsOpened, setConsentModalIsOpened] = useState(false);
-  const [isCreateAnalysisModalOpened, setIsCreateAnalysisModalOpened] =
-    useState(false);
-
-  const closeCreateAnalysisModal = () => {
-    setIsCreateAnalysisModalOpened(false);
-  };
-
-  const closeConsentModal = () => {
-    setConsentModalIsOpened(false);
-  };
-
-  const openConsentModal = () => {
-    setConsentModalIsOpened(true);
-    updateSearchParams({ customFieldId });
-  };
-
-  const onAcceptConsent = () => {
-    setConsentModalIsOpened(false);
-    setIsCreateAnalysisModalOpened(true);
-  };
-
-  const relevantAnalyses =
+  const relevantAnalysis =
     analyses?.data &&
-    analyses?.data?.filter((analysis) =>
+    analyses?.data?.find((analysis) =>
       analysis.relationships.custom_fields.data.some(
         (field) => field.id === customFieldId
       )
     );
-  const hasAnalyses = relevantAnalyses && relevantAnalyses.length > 0;
+
+  // Create an analysis if there are no analyses yet
+  useEffect(() => {
+    if (analyses && customFieldId && !relevantAnalysis) {
+      addAnalysis({
+        projectId: phaseId ? undefined : projectId,
```

Comment:
Since we'll always have phases from now on, do you think we should change the `addAnalysis` to only accept a `phaseId` and `customFieldIds`?

Reply:
It's a bit more complicated than this, unfortunately because for ideation we analyse all the ideas in a project regardless of the phase and for a survey we only analyse the phase. So we can't change it, at least for now. Maybe moving forward, we can rethink.